packages/frontend/web-ui/src/auth/hooks/useResetPassword.ts

```diff
@@ -0,0 +1,316 @@
+    });
+  }
+
+  function buildValidation(): ResetPasswordFormValidationResult {
+    const formValidation: ResetPasswordFormValidationResult = {};
+
+    const passwordValidation: Either<string, undefined> = validatePassword(
+      resetPasswordData.form.fields.password,
+    );
+
+    if (!passwordValidation.isRight) {
+      formValidation.password = passwordValidation.value;
+    } else {
+      userMeUpdateQueryV1.password = resetPasswordData.form.fields.password;
+    }
+
+    const confirmPasswordValidation: Either<string[], undefined> =
+      validateConfirmPassword(
+        resetPasswordData.form.fields.password,
+        resetPasswordData.form.fields.confirmPassword,
+      );
+
+    if (!confirmPasswordValidation.isRight) {
+      formValidation.confirmPassword =
+        confirmPasswordValidation.value.join(' ');
+    }
+
+    return formValidation;
+  }
+
+  function createAuth(): void {
+    if (codeParam !== null) {
+      setResetPasswordData({
+        form: {
+          fields: {
+            ...resetPasswordData.form.fields,
+          },
+          validation: { ...resetPasswordData.form.validation },
+        },
+        status: UseResetPasswordStatus.creatingAuth,
+      });
+
+      void triggerCreateAuthV2({
+        authCreateQuery: {
+          code: codeParam,
+          kind: 'code',
+        },
+      });
+    } else {
+      setResetPasswordData({
+        form: {
+          errorMessage: UNEXPECTED_AUTH_ERROR_MESSAGE,
+          fields: {
+            ...resetPasswordData.form.fields,
+          },
+          validation: { ...resetPasswordData.form.validation },
+        },
+        status: UseResetPasswordStatus.backendError,
+      });
+    }
+  }
+
+  function updateUserMe(): void {
+    setResetPasswordData({
+      form: {
+        fields: {
+          ...resetPasswordData.form.fields,
+        },
+        validation: { ...resetPasswordData.form.validation },
+      },
+      status: UseResetPasswordStatus.updatingUser,
+    });
+
+    void triggerUpdateUserMeV1({
+      params: [userMeUpdateQueryV1],
+    });
+  }
+
+  function onSubmit(event: React.FormEvent): void {
+    event.preventDefault();
+
+    if (resetPasswordData.status !== UseResetPasswordStatus.pending) {
+      return;
+    }
+
+    const formValidation: ResetPasswordFormValidationResult = buildValidation();
+
+    if (Object.keys(formValidation).length === 0) {
+      setResetPasswordData({
+        form: {
+          fields: {
+            ...resetPasswordData.form.fields,
+          },
+          validation: formValidation,
+        },
+        status: resetPasswordData.status,
+      });
+
+      updateUserMe();
+    } else {
+      setResetPasswordData({
+        form: {
+          fields: {
+            ...resetPasswordData.form.fields,
+          },
+          validation: formValidation,
+        },
+        status: UseResetPasswordStatus.pending,
+      });
+    }
+  }
+
+  function buildErrorMessage(
```

Comment:
We should split this in two functions, one for each endpoint involved

Reply:
Done!